### ISSUE TITLE
Popup hit boxes in wrong location with a window stretch mode enabled

### ISSUE THREAD

**Issue by voylin** [godotengine/godot#59610]
### Godot version

4.0.alpha5

### System information

Arch (Linux-lts 5.15.30-1)

### Issue description

So I load an instance of a scene with an OptionButton.
But when I want to click on options from a MenuButton or OptionMenu, it fails to register. After a lot of testing and trying I finally figured out what the problem is.

If you don't put the window in fullscreen of let the window take up all the space on the screen, than the hitboxes for the popup are in the wrong location, as can be seen in my screenshot. My mouse is off to the right where the hitboxes would be if it would be in full window.

![IMG_20220328_104732635](https://user-images.githubusercontent.com/56970759/160313464-a0cce2f1-afbb-4ee5-98a9-2ea393299634.jpg)

Now, I am not certain if this is a tiled window manager issue or a general issue, but it still does not work even if I put it in floating window mode

### Steps to reproduce

Edit 2022-03-28 12:34:
Create an OptionButton or MenuButton and add it as a child through scripts.

### Minimal reproduction project
[TestRoom.tar.gz](https://github.com/godotengine/godot/files/8359446/TestRoom.tar.gz)

**Comment by Rindbee** [comment]
It looks like i3-wm, I use it too.

But I'm sorry, I can not reproduce this issue in i3-wm. Maybe you should provide a small project with this issue.

**Comment by voylin** [comment]
> It looks like i3-wm, I use it too.
> 
> But I'm sorry, I can not reproduce this issue in i3-wm. Maybe you should provide a small project with this issue.

Sorry about that, it took me some time to replicate it. It appears to only happen when loading in a scene with an option button. I attached a small project that has this issue.

**Comment by Rindbee** [comment]
In your provided project, the problem seems to be caused by the modified `display/window/stretch/mode` in the ProjectSettings.

**Comment by bruvzg** [comment]
The demo is running with embedded subwindows enabled, so it is not dependent on Window manager and platform. I can reproduce it on macOS as well, but only after the window is resized.

> In your provided project, the problem seems to be caused by the modified display/window/stretch/mode in the ProjectSettings.

Seems like it is the case, probably `strech mode` is not taken into account by `window` transform for embedded windows.

**Comment by voylin** [comment]
Yep, that's indeed the culprit. The moment I select any of those modes the problem happens. I do this mainly for UI Scaling, so fine for now but this is really a bug which should be fixed.

I think this has to do that the popup is part of the window node?

**Comment by Sauermann** [comment]
Looks like a duplicate of #59573.
It will get fixed by #59118 or #59310.
